lib/modules/manager/hermit/extract.ts

```diff
@@ -0,0 +1,103 @@
+import is from '@sindresorhus/is';
+import minimatch from 'minimatch';
+import upath from 'upath';
+import { logger } from '../../../logger';
+import { readLocalDirectory } from '../../../util/fs';
+import { regEx } from '../../../util/regex';
+import { HermitDatasource } from '../../datasource/hermit';
+import type { PackageDependency, PackageFile } from '../types';
+import type { HermitListItem } from './types';
+
+const pkgReferenceRegex = regEx(`(?<packageName>.*?)-(?<version>[0-9]{1}.*)`);
+
+/**
+ * extractPackageFile scans the folder of the package files
+ * and looking for .{packageName}-{version}.pkg
+ */
+export async function extractPackageFile(
+  content: string,
+  filename: string
+): Promise<PackageFile | null> {
+  logger.trace('hermit.extractPackageFile()');
+  const dependencies = [] as PackageDependency[];
+  const packages = await listHermitPackages(filename);
+
+  if (packages === null) {
+    logger.warn(`error listing hermit packages`);
+    return null;
+  }
+
+  for (const p of packages) {
+    // version of a hermit package is either a Version or a Channel
+    // Channel will prepend with @ to distinguish from normal version
+    const version = p.Version === '' ? `@${p.Channel}` : p.Version;
+
+    const dep: PackageDependency = {
+      datasource: HermitDatasource.id,
+      depName: p.Name,
+      currentValue: version,
+    };
+
+    dependencies.push(dep);
+  }
+
+  return { deps: dependencies };
+}
+
+/**
+ * listHermitPackages will fetch all installed packages from the bin folder
+ */
+async function listHermitPackages(
+  hermitFile: string
+): Promise<HermitListItem[] | null> {
+  logger.trace('hermit.listHermitPackages()');
+  const hermitFolder = upath.dirname(hermitFile);
+
+  let files: string[] = [];
+
+  try {
+    files = await readLocalDirectory(hermitFolder);
+  } catch (e) {
+    logger.warn(
```

Comment:
```suggestion
    logger.debug(
```

We tend to not warn here, as it's a system fault, so renovate should retry automatically